### PR TITLE
Reset min_max normalization values at the start of validation epoch

### LINKED ISSUE #2027: [Bug]: Not very useful for inference: "min", "max" and "image_threshold"  stored in metadata during training
### Describe the bug

I was computing some inferences with my trained models and I susprised when I saw that **the min and max values stored during training didn't match with the values I was seeing during my inference.** I mean, based on anomalib's code, those "min" and "max" (which corresponds to the `src.anomalib.utils.metrics.min_max.py.MinMax` metric) the min and max are, respectively, the min and max values of the anomaly maps of the validation calculated during training. So, if my validation set is the same, why the metdata's stored "max" is much higher than the maximum of the anomaly maps I was inferencing? And the same with "min" and also, with "image_threshold".

So I just realized that it's normal that the "max" stored in metadata is higher than real "max" of anomaly maps with the already trained model, and "min" is much lower. It's because of the way of calculating those values is updating them if you see an anomaly map with higher values than stored "max" or lower than stored "min" (`self.max = torch.max(self.max, torch.max(predictions))`) and **in the firsts epochs, when the network is not yet very fit, the "max" would be very high and the "min" very low, thus making it very difficult to update these values again and leaving us at the end of training using values calculated during the first epochs of an "early model". **

The point here is that metadata's "min", "max", and "image_threshold" when not masks are provided (in this case the image threshold would be the max of the good anomaly maps) are not so useful in the inferences of the already trined model. These values are used normally within the `post_process` step inside the `predict` of the `Inferencers` like `TorchInferencer`.

Maybe **should they reset at each epoch_start**??

Thank you so much for your valuable work!!

### Dataset

MVTec

### Model

N/A

### Steps to reproduce the behavior

- Train an anomalib model
- Perform inference with the trained model in the same validation set that was used during trainig to validate and compute the metrics
- See how "min", "max" and (if not masks provided for anoamalous classes) "image_threshold" don't match with the anomaly maps you are inferencing

### OS information

OS information:
- OS: Windows 10
- Python version: 3.10.14
- Anomalib version: 1.0.1
- PyTorch version: 2.2.0+cu121
- CUDA/cuDNN version: 12.4/8.9.2.26
- GPU models and configuration: 1x GeForce RTX 3090

### Expected behavior

To min", "max" and (if not masks provided for anoamalous classes) "image_threshold" match with the anomaly maps you are inferencing with an anomalib's trained model.

### Screenshots

_No response_

### Pip/GitHub

GitHub

### What version/branch did you use?

_No response_

### Configuration YAML

```yaml
Not relevant
```

### Logs

```shell
Not relevant
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #2139: [Bug]: EfficientAd strange anomaly map results
### Describe the bug

I know there was a PR that supposedly fixed generl performance problems with EfficientAd model (#2015).
However, I am trying (with latest anomalib version) to train **EfficientAd on MVTec bottle and I have found several problems.**

I relate them sepparately below:

- **From one hand**, the model **appears to work well but it gives some strange "circles"** as anomalies on the borders of the images. 
image
even in good samples
even in good samples

- **From another hand**, it sepparates perfectly the distributions (conf matrix perfect) but it seems to have wrong scales, like too sparse (which I think it can be due to #2027), maybe a sigmoid function would fix this?, proofs following:
image

- Also, and I think it is related with the previous point (abaout scales when normalizing), even that the segmentation map looks okay (despite the "border circles"), the anomaly map looks that have always same value constant (#1647) 
image
As said in #1647 I tried both `normalization_metod = min_max` ( above results) and `normalization_method = none`, giving below results: 
image
It looks like the map is being normalized with negative values or something because the more blue pixels are the segmented ones (when the expected behaviour is that the red ones to be the most anomalous). It is strange. As a note: the dataset normalization is retrieved from EfficientAd default transforms (none because in forward pass there is already imagenet norm).

- I have tried all possible combinations of `pad_maps` and `padding` hyperparameters. The best are the default ones (`pad_maps = True`. `padding = False`). 

### Dataset

MVTec

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

Train EfficientAd on MVTec bottle with default configurations.

### OS information

OS information:
- OS: Windows 10 Pro
- Python version: 3.10.14
- Anomalib version: 1.2.0dev
- PyTorch version: 2.2.0+cu121
- CUDA/cuDNN version: 12.5/8.9.7.29
- GPU models and configuration: 1x GeForce RTX 3090
- Any other relevant information: No

### Expected behavior

EfficientAd to work well in MVTec

### Screenshots

_No response_

### Pip/GitHub

pip

### What version/branch did you use?

1.2.0dev

### Configuration YAML

```yaml
data:
  class_path: anomalib.data.MVTec
  init_args:
    root: data/external/MVTec
    category: bottle
    train_batch_size: 1
    eval_batch_size: 1
    num_workers: 8
    task: segmentation
    transform: null
    train_transform: null
    eval_transform: null
    test_split_mode: from_dir
    test_split_ratio: 0.2
    val_split_mode: same_as_test
    val_split_ratio: 0.5
    seed: null
model:
  class_path: anomalib.models.EfficientAd
  init_args:
    imagenet_dir: data/external/imagenette
    teacher_out_channels: 384
    model_size: S
    lr: 0.0001
    weight_decay: 1.0e-05
    padding: false
    pad_maps: true
normalization:
  normalization_method: none
metrics:
  image:
  - F1Score
  - AUROC
  pixel:
  - F1Score
  - AUROC
  threshold:
    class_path: anomalib.metrics.F1AdaptiveThreshold
    init_args:
      default_value: 0.5
logging:
  log_graph: false
seed_everything: 6120
task: segmentation
default_root_dir: results
ckpt_path: null
trainer:
  accelerator: auto
  strategy: auto
  devices: 1
  num_nodes: 1
  precision: 32
  logger:
  - class_path: anomalib.loggers.AnomalibWandbLogger
    init_args:
      project: mvtec-bottle
  - class_path: anomalib.loggers.AnomalibMLFlowLogger
    init_args:
      experiment_name: mvtec-bottle
  callbacks:
  - class_path: anomalib.callbacks.checkpoint.ModelCheckpoint
    init_args:
      dirpath: weights/lightning
      filename: best_model-{epoch}-{image_F1Score:.2f}
      monitor: image_F1Score
      save_last: true
      mode: max
      auto_insert_metric_name: true
  - class_path: lightning.pytorch.callbacks.EarlyStopping
    init_args:
      patience: 5
      monitor: image_F1Score
      mode: max
  fast_dev_run: false
  max_epochs: 1000
  min_epochs: null
  max_steps: 70000
  min_steps: null
  max_time: null
  limit_train_batches: 1.0
  limit_val_batches: 1.0
  limit_test_batches: 1.0
  limit_predict_batches: 1.0
  overfit_batches: 0.0
  val_check_interval: 1.0
  check_val_every_n_epoch: 1
  num_sanity_val_steps: 0
  log_every_n_steps: 50
  enable_checkpointing: true
  enable_progress_bar: true
  enable_model_summary: true
  accumulate_grad_batches: 1
  gradient_clip_val: 0
  gradient_clip_algorithm: norm
  deterministic: false
  benchmark: false
  inference_mode: true
  use_distributed_sampler: true
  profiler: null
  detect_anomaly: false
  barebones: false
  plugins: null
  sync_batchnorm: false
  reload_dataloaders_every_n_epochs: 0
  default_root_dir: null
```

### Logs

```shell
No
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by abc-125**
## 📝 Description

Currently, min_max values for normalization are updated every validation epoch. This update keeps irrelevant values from previous epochs by comparing new and old ones:
`self.max = torch.max(self.max, torch.max(predictions))`

It seems like it does not affect most of the models, but with at least EfficientAD, the final anomaly maps after training are normalized the wrong way (see #2139 or this [discussion](https://github.com/openvinotoolkit/anomalib/discussions/1647)).

This fix resets min_max values at the beginning of every validation epoch to ensure that only values from the recent one will be used for normalization.

🛠️ Fixes #2027, thank you @CarlosNacher for pointing this problem out! Partially fixes #2139.

## ✨ Changes

Select what type of change your PR is:

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

**Comment by abc-125** [comment]
Sorry, this was already solved in this [PR](https://github.com/openvinotoolkit/anomalib/pull/2130).
